helper/triples.php

```diff
@@ -39,6 +40,40 @@ function _expandTokens($str) {
     }
 
     /**
+     * Initializes SQL filter of blocked graphs (no page permission by ACL rules).
+     */
+     function init_blocked_graphs() {
+      if ($this->getConf('use_acl') and self::$blocked_graphs_sql == '') {
+        $sql = "SELECT DISTINCT graph FROM " . self::$readable;
+        
+        // prepare query
+        $query = $this->_db->prepare($sql);
+        if($query == false) return;
+
+        // execute query
+        $res = $query->execute($values);
+        if($res === false) {
+            $error = $query->errorInfo();
+            msg(sprintf($this->getLang('error_graphs_fetch'),hsc($error[2])),-1);
+        }
+
+        // fetch results and return them
+        $result = $query->fetchAll(PDO::FETCH_ASSOC);
+        $query->closeCursor();
+        
+        // a list of graphs reachable by the current user is generated 
+        $blocked_graphs = array();
+        foreach ($result as $row) {
+          if (auth_quickaclcheck($row['graph']) < AUTH_READ)
+            array_push($blocked_graphs, "'" . $row['graph'] . "'");
```

Comment:
The `$row['graph']` should be escaped for the correct database. While it is not easily possible to get unexpected content into the `graph` column of the database, it is not unheard of for external tools to add additional graphs beyond those from the wiki.

The use of [PDO::quote](http://php.net/manual/en/pdo.quote.php) is very useful here.